### DISCUSSION TITLE
Idea: put jsvm .d.ts on DefinitelyTyped?

### DISCUSSION THREAD

**Discussion by benallfree** [pocketbase/pocketbase#5512] [Q&A] [answered]
Circling back to #3360, I was wondering about creating a `@types/pocketbase-jsvm` and somehow enhancing it to reflect goja's full environment. Something similar to @types/node. Would this be something that would make sense to discuss with the goja project as well? Maybe a @types/goja and @types/goja-node, and then @types/pocketbase-jsvm builds upon that?

**Comment by ganigeorgiev** [comment]
I don't think it makes sense because neither PocketBase nor goja technically operates with TypeScript out of the box.

The PocketBase JSVM ambient TS declarations are auto generated based on the specific PocketBase bindings, meaning that they can change from one version to another. Publishing it to the DefinitelyTyped repo at the moment is not feasible as it will be too much hassle to maintain.

**Reply by benallfree** [reply]
Would you mind if I created an unofficial `@types/pocketbase-jsvm` package? I could add you as an owner as well, in case you ever decide to do something with it. For me Typescript is valuable in the toolchain for creating robust PB JSVM code. (*or suggest a different package name if that one would confuse users*)

**Reply by ganigeorgiev** [reply, marked as answer]
There is no need to add me as "owner" and I don't mind publishing such package as long as it is clear that it is not officially supported by me.

**Reply by benallfree** [reply]
(PS - `@types/*` are auto-loaded in most environments, which is why it would be nice to have it there as opposed to somewhere else)

**Reply by ganigeorgiev** [reply]
I understand, but as mentioned earlier the bindings can change from one PocketBase version to another (and there is no backward compatibility yet), aka. you usually will want the `pb_hooks` ambient types to represent only what is available by the current running PocketBase executable.

**Comment by benallfree** [comment]
I posted https://github.com/dop251/goja/issues/608 as well.
